proto/examples/sasltest/main.rs

```diff
@@ -0,0 +1,167 @@
+use core::panic;
+use futures_util::future::err;
+use sspi::builders::EmptyInitializeSecurityContext;
+use sspi::AuthIdentity;
+use sspi::ClientRequestFlags;
+use sspi::CredentialUse;
+use sspi::DataRepresentation;
+use sspi::Ntlm;
+use sspi::SecurityBuffer;
+use sspi::SecurityBufferType;
+use sspi::SecurityStatus;
+use sspi::Sspi;
+use sspi::SspiImpl;
+use sspi::Username;
+use std::env;
+use std::net::SocketAddr;
+use std::str::FromStr;
+use tokio::net::TcpStream;
+use tokio_util::codec::Framed;
+use tracing::Level;
+
+use futures_util::sink::SinkExt;
+use futures_util::stream::StreamExt;
+
+use ldap3_proto::proto::*;
+use ldap3_proto::LdapCodec;
+
+#[tokio::main]
+async fn main() -> Result<(), ()> {

```

Comment:
Suggestion: you could add `anyhow` as a dev dependency so that you write this:
```suggestion
async fn main() -> anyhow::Result<()> {

```

```rust
use anyhow::Context as _;

let stream = TcpStream::connect(addr)
    .await
    .with_context(|| format!("failed to connect to {addr}"))?;

tracing::subscriber::set_global_default(subs).context("failed to set global tracing subscriber")?;
```

(Maybe wait for @Firstyear to see if they are okay with that)